src/phpDocumentor/Uri.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor;
+
+final class Uri
+{
+    /**
+     * @var string
+     */
+    private $uri;
+
+    public function __construct($uri)
+    {
+        $this->uri = (string)$uri;
+
+        $this->validate($uri);
```

Comment:
It is common to validate a value before handling it (in this case assigning it to the property); this is because you want to fail early as this will improve performance and because handling an invalid variable might give unexpected issues. In this specific instance it will not cause an issue but it is a good thing to get into your system